pghoard/pghoard.py

```diff
@@ -566,6 +632,19 @@ def handle_site(self, site, site_config):
 
         metadata = self.get_new_backup_details(site=site, site_config=site_config)
         if metadata and not os.path.exists(self.config["maintenance_mode_file"]):
+            if site in self.delta_backup_failures:
+                retries = self.delta_backup_failures[site].retries
+                if retries > site_config["basebackup_delta_mode_max_retries"]:
+                    self.log.info("Giving up backup after exceeding max retries: %r", retries)
+                    return
+                else:
+                    if utc_now(
+                    ) >= self.delta_backup_failures[site].last_failed_time + datetime.timedelta(seconds=2 ** retries):
```

Comment:
This ends up adding very little delay. 1 second the first time, 2 seconds the second time, and so on. I think something like 30 seconds first time, 2 minutes second, 5 minutes third, 15 minutes fourth etc. with maybe a cap at one hour would be more sensible.